v2/internal/runner/inputs.go

```diff
@@ -63,28 +67,22 @@ func (r *Runner) initializeTemplatesHTTPInput() (*hybrid.HybridMap, error) {
 }
 
 var (
-	drainReqSize = int64(8 * 1024)
-	httpSchemes  = []string{"https", "http"}
+	httpSchemes = []string{"https", "http"}
 )
 
 // probeURL probes the scheme for a URL. first HTTPS is tried
 // and if any errors occur http is tried. If none succeeds, probing
 // is abandoned for such URLs.
-func probeURL(input string, httpclient *retryablehttp.Client) string {
+func probeURL(input string, httpxclient *httpx.HTTPX) string {
 	for _, scheme := range httpSchemes {
 		formedURL := fmt.Sprintf("%s://%s", scheme, input)
-		req, err := retryablehttp.NewRequest(http.MethodGet, formedURL, nil)
+		req, err := httpxclient.NewRequest(http.MethodGet, formedURL)
```

Comment:
What do you think about using `HEAD` as we need to check if there is a web server (and skip body elaboration)?